src/providers/postgres/qgspostgreslistener.cpp

```diff
@@ -58,7 +61,43 @@ QgsPostgresListener::~QgsPostgresListener()
 void QgsPostgresListener::run()
 {
   PGconn *conn = nullptr;
-  conn = PQconnectdb( mConnString.toLocal8Bit() );
+  QString connectString = mConnString;
+
+  connectString += QStringLiteral( " connect_timeout=%1" ).arg( PG_CONNECT_TIMEOUT );
+  conn = PQconnectdb( connectString.toUtf8() );
+
+  if ( PQstatus( conn ) != CONNECTION_OK )
+  {
+    QgsDataSourceUri uri( connectString );
+    QString username = uri.username();
+    QString password = uri.password();
+
+    PQfinish( conn );
+
+    QgsCredentials::instance()->lock();
+
+    if ( QgsCredentials::instance()->get( mConnString, username, password, PQerrorMessage( conn ) ) )
```

Comment:
Why are you not starting by this check (are there credentials?) and otherwise trying without credentials?

Reply:
I tried to keep the first database connection request as similar as possible to the request used in the original code. In my opinion, this will preserve the behavior of QGIS for users who are already using Postgres NOTIFY Listener in conjunction with SSL certificates (without username and password) to connect to the database. Unlike the original, the encoding was changed and a timeout was added for connecting to the database.

Reply:
@benoitdm-oslandia I use it 
https://github.com/qgis/QGIS/blob/89c069f695a0684e9e1e2df60fbc1c9b8f57b1fe/src/providers/postgres/qgspostgresconn.cpp#L213
as a reference code.
I'm not sure that the implementation proposed in this PR will allow you to connect with SSL certificates, but I think that it will definitely not get worse than it is at the moment...

Reply:
ok